probert/storage.py

```diff
@@ -147,44 +154,49 @@ class Storage():
         present.
     """
     probe_map = {
-        'bcache': bcache.probe,
-        'blockdev': blockdev_probe,
-        'dasd': dasd.probe,
-        'dmcrypt': dmcrypt.probe,
-        'filesystem': filesystem.probe,
-        'lvm': lvm.probe,
-        'mount': mount.probe,
-        'multipath': multipath.probe,
-        'os': os.probe,
-        'raid': raid.probe,
-        'zfs': zfs.probe
+        'bcache': Probe(bcache.probe),
+        'blockdev': Probe(blockdev_probe),
+        'dasd': Probe(dasd.probe),
+        'dmcrypt': Probe(dmcrypt.probe),
+        'filesystem': Probe(filesystem.probe),
+        'lvm': Probe(lvm.probe),
+        'mount': Probe(mount.probe),
+        'multipath': Probe(multipath.probe),
+        'os': Probe(os.probe, False),
+        'raid': Probe(raid.probe),
+        'zfs': Probe(zfs.probe),
     }
 
     def __init__(self, results={}):
         self.results = results
         self.context = pyudev.Context()
 
-    def _get_probe_types(self):
-        return {ptype for ptype, pfunc in self.probe_map.items() if pfunc}
+    def _get_probe_types(self, all=False):
+        return {ptype for ptype, probe in self.probe_map.items()
+                if probe.pfunc and (probe.in_default_set or all)}
```

Comment:
Is there any way probe.pfunc can be non-truthy here? I never understood this bit of the code :)

Reply:
As far as I can tell it's always going to be true.  This reads like a C-ism, and if we wanted this then `getattr(foo, 'pfunc')` is probably what was desired.  I'm just removing the pfunc reference.